Loop/Managers/LoopAppManager.swift

```diff
@@ -622,14 +622,7 @@ class LoopAppManager: NSObject {
             settingsViewModel: settingsViewModel
         )
 
-        let statusTableView = StatusTableView(viewModel: viewModel)
-            .environmentObject(deviceDataManager.displayGlucosePreference)
-            .environment(\.appName, Bundle.main.bundleDisplayName)
-            .environment(\.isInvestigationalDevice, FeatureFlags.isInvestigationalDevice)
-            .environment(\.loopStatusColorPalette, .loopStatus)
-            .environment(\.settingsManager, settingsManager)
-            .environment(\.temporaryPresetsManager, temporaryPresetsManager)
-            .edgesIgnoringSafeArea(.top)
+        let statusTableView = StatusTableView(viewModel: viewModel, displayGlucosePreference: deviceDataManager.displayGlucosePreference, settingsManager: settingsManager, temporaryPresetsManager: temporaryPresetsManager)
```

Comment:
Moving environment modifiers into the view itself make type matching the type-erased viewController much easier.

Reply:
Honestly not a huge fan of this. I understand why you're doing it; so you can cast it later, but having to cast also seems like a smell.

It seems like the deep link manager should probably already have a reference to the the status table view controller, rather than having to dig through the view stack, and unwrap/cast things.

It looks like LoopAppManager.handle is the entry point, passing the url off to the deeplinkManager. LoopAppManager also creates StatusTableView, can we get a ref to the status table view controller through there?

Reply:
It seems like all DeeplinkManager is pass through, so I don't think that really justifies another manager layer.  Same with RootNavigationController.  LoopAppManager/AppDelegate is the main entry point; and the StatusTableViewController has the real logic, and LoopAppManager should be able to retain a ref to StatusTableViewController/StatusTableView

Reply:
Yes, I think that's a lot better. Thanks!